### DISCUSSION TITLE
Where to install the IDL Astronomy User’s Library

### DISCUSSION THREAD

**Discussion by AngelDGJurado** [gnudatalanguage/gdl#1657] [Q&A]
Hello, I am not used to IDL or GDL, but I need to use it for my thesis, but I don't know where to put the "astron" library so that it is recognized in the GDL workbench. I hope someone can help me with this request, I am new to this.
English is not my main language so sorry for any issues.

**Comment by alaingdl** [comment]
English is clearly not my first language too :sweat_smile: 

You have several solutions for that. And we have two universes : MSwin (I know nothing) and OSX + Linux

You can set it before starting GDL, or after.
 
What I have is an alias, a Bash file called LastGDL, to start GDL adding several third party libs. (Astron, mpfit, Coyote, my libs ...)

```
$ more LastGDL (file in 755 mode + in a path for executables)

ASTRO_LIB='+'$AC_HOME'/IDLetGDL/Librairies/idlastro/:'
COYOTE_LIB=$AC_HOME'/IDLetGDL/Librairies/coyote/:'
GDL_PATH='+'$HOME'/GDL/last-gdl/src/pro/:'
[...] others lines
#
GDL_PATH=$GDL_PATH$CMSV_LIB
GDL_PATH=$GDL_PATH$ASTRO_LIB
GDL_PATH=$GDL_PATH$IDL_LIB
GDL_PATH=$GDL_PATH$MPFIT_LIB
[...] others lines
#
export GDL_PATH=$GDL_PATH
#
$GDL_ROOT/last-gdl/install/bin/gdl $@
[...] $@ is a trick to be able to have arguments at startup time

```
just cut the lines you don't need and adjust the paths ...

Another way, assuming your GDL is well install with access to the PRO files we have, is to do : 
```
GDL> path_add, '/your/path/to/astronlib/', /rec
```
(first / for absolute path ...)

Later you can also add this line to a GDL startup file ...

HTH

Questions welcome

Alain

**Comment by AngelDGJurado** [comment]
I forgot to add, I'm in windows 10 if that matters.
I don't know how to do the first option, so i tried the second option "path_add, '/your/path/to/astronlib/', /rec" in the GDL Workbench and i got the message "No explicit position : the defaut is AT THE BEGINNING", and after that everything was the same, it still doesn't recognise the astron file.
That file is in the same PATH library that the GDL uses, but i don't know where to put it exactly

**Comment by GillesDuvert** [comment]
look at [this answer](http://www.idlcoyote.com/code_tips/installcoyote.php#WINDOWS_INSTALL). It concerns mainly another package, `coyotelib`, but 1) you'll probably need it anyway and 2) it's the same for the astron library. Read carefully as it is not as easy on windows as on linux.
BTW, @alaingdl probably we need to investigate this path_add procedure problem?

**Comment by alaingdl** [comment]
concerning the way to have a simple procedure to have it "easily for beginners" I do have somewhere a draft for IPM made by a student but I do not consider it as OK now.

concerning PATH_ADD, I never try on MSwin. Could you look at the internal !PATH before and after ?
```
HELP,/path
path_add, '/your/path/to/astronlib/', /rec
HELP,/path
```
the second one should have been changed ...

If using the absolute path is a problem, you can do that relatively.
(since it is in MSwin, *it should* but I gave no recent experience *it does* !)

**Comment by GillesDuvert** [comment]
@AngelDGJurado additionnally the GDL workbench is, as far as I know, untested under Windows.
Please notice that directory separator on Windows is "\" not "/" when using path_add.

**Comment by AngelDGJurado** [comment]
Ok, for what i can figure out it's working as it should, it recognises the astron folder in path 

> concerning the way to have a simple procedure to have it "easily for beginners" I do have somewhere a draft for IPM made by a student but I do not consider it as OK now.
> 
> concerning PATH_ADD, I never try on MSwin. Could you look at the internal !PATH before and after ?
> 
> ```
> HELP,/path
> path_add, '/your/path/to/astronlib/', /rec
> HELP,/path
> ```
> 
> the second one should have been changed ...
> 
> If using the absolute path is a problem, you can do that relatively. (since it is in MSwin, _it should_ but I gave no recent experience _it does_ !)

Ok, as far as I can understand, it works, the path is added to the top of the list, however the problem I originally had is still present (I thought it was because I didn't install it correctly, but it seems as if that wasn't the problem.
It looks like that wasn't the problem haha, but thanks for all the help anyway
